fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -1399,4 +1405,35 @@ void testQueryWithManyDocuments() {
             assertPrimaryKeys("text:morningstart", false, Set.of());
         }
     }
+
+    @ParameterizedTest
+    @RandomSeedSource
+    void encrypted(long seed) throws Exception {
+        final RollingTestKeyManager rollingTestKeyManager = new RollingTestKeyManager(seed);
+        keyManager = rollingTestKeyManager;
+        // Save in many transactions to get more segments and so more blocks.
+        for (TestRecordsTextProto.SimpleDocument doc : DOCUMENTS) {
+            try (FDBRecordContext context = openContext()) {
+                openRecordStore(context);
+                recordStore.saveRecord(doc);
+                commit(context);
```

Comment:
This should be merging in-line when you commit. Would it be worthwhile to test merging separately? Perhaps that's best covered by LuceneIndexMaintenanceTest.